### PR TITLE
Automate docs site pr

### LINKED ISSUE #1866: Metadata: Automate release note PRs to docs site
### Description
Create a task that automatically generates release note PRs to the docs site. The node team has already implemented this and may be a good resource to explore during the work on this ticket.

### Acceptance Criteria
PRs to docs site are automatically generated that include: 
* full release note content 
* metadata field content

### PR DESCRIPTION
**Pull request by hannahramadan**
Introduces automatic release note generation and pull requests to newrelic/docs-website.

- `generate_release_notes.rb` uses information in CHANGELOG to build release notes
- `release_notes.yml` runs methods from `generate_release_notes.rb` to generate the release notes and the branch name, and then uses that information to submit a PR to docs-website on pushes to the agent's `main` branch.

Closes #1866

### CODE REVIEW

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [inline .github/workflows/release_notes.yml:21]: Creating environment variables is how we add dynamic naming for the source file and branch names.

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [inline .github/workflows/release_notes.yml:27]: This step is separate from the above since environment variables can only be used outside the step in which they are created.

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:24]: What if the `write_output_file` method returned the filename that it wrote to instead of the result of the write operation? Could you combine the two separate "write file" and "get filename" operations into one if you did that?

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:9]: These can go away

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:38]: Are all of the values in this section the final merge ready values?

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:42]: Should this have a space?

```suggestion
          commit_message: 'chore (ruby agent): add release notes'
```

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:68]: Could you walk me through what these 3 `git` commands do here?

File: .github/workflows/scripts/generate_release_notes.rb
Comment (fallwith) [inline .github/workflows/scripts/generate_release_notes.rb:72]: For consistency's sake, I think:

a) This line should be `puts write_filename` to use `puts` instead of `p`
b) The naming of "filename" vs "file_name" should be reconciled

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [reply .github/workflows/release_notes.yml:42]: It doesn't look like the docs team usually does spaces!

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [reply .github/workflows/release_notes.yml:68]: Ooo thanks for calling this out. I think we might only need `git push origin --delete --force ruby_release_notes` to delete the remote branch. I'll test this and update it.

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [reply .github/workflows/release_notes.yml:38]: Nope! Those will still need to be updated once we have bot approval.

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [reply .github/workflows/release_notes.yml:27]: I haven't fully understood why, but yes. Trying to build `source_file` directly using `${{env.file_name}}` makes changes in a GitHub workspace. We were only able to get this to build correctly by first adding the extra step to build the correct file name

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [reply .github/workflows/release_notes.yml:24]: I think we could! Do you think that would muddy the human readability of what is happening when it comes to assigning the Github variables?

Comment (fallwith) [reply .github/workflows/release_notes.yml:22]: I think the storing of the result into an env var named "file_name" still makes it clear what's happening:

```shell
echo "file_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.write_output_file)" >> $GITHUB_ENV
```

File: .github/workflows/release_notes.yml
Comment (fallwith) [reply .github/workflows/release_notes.yml:25]: Gotcha. So my follow up question is... why do you need to create the file in the other step in the first place? Could you just move the file creation and name fetching to this section? That way you wouldn't need to create a new env var that doesn't do anything but reference the other step's env var.

Comment (hannahramadan) [reply .github/workflows/release_notes.yml:23]: Resolved synchronously with James. These steps are now combined. 42a4a2b10ba783a592233a272a0c29266c639964

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:20]: I'd prefer the relative path here if it still works:

```shell
ruby -r .github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.write_output_file
```

File: .github/workflows/release_notes.yml
Comment (hannahramadan) [reply .github/workflows/release_notes.yml]: Discussed live. There is more work to be done on a job that deletes the branch upon Action failure. This work has been summarized in [#1919](https://github.com/newrelic/newrelic-ruby-agent/issues/1919)

File: .github/workflows/release_notes.yml
Comment (fallwith) [inline .github/workflows/release_notes.yml:21]: Same curiosity about the relative path here.

Comment (hannahramadan) [reply .github/workflows/release_notes.yml:21]: Only full path seems to work here

**Review by fallwith — APPROVED** [review]

**Review by fallwith — APPROVED** [review]

**Review by kaylareopelle — APPROVED** [review]

**Pull request merged**
